src/pkg/packager/deploy.go

```diff
@@ -138,20 +138,10 @@ func (p *Packager) Deploy(ctx context.Context) error {
 
 // deployComponents loops through a list of ZarfComponents and deploys them.
 func (p *Packager) deployComponents(ctx context.Context) (deployedComponents []types.DeployedComponent, err error) {
-	// Check if this package has been deployed before and grab relevant information about already deployed components
-	if p.generation == 0 {
-		p.generation = 1 // If this is the first deployment, set the generation to 1
-	}
-
 	// Process all the components we are deploying
 	for _, component := range p.cfg.Pkg.Components {
-		deployedComponent := types.DeployedComponent{
-			Name:               component.Name,
-			Status:             types.ComponentStatusDeploying,
-			ObservedGeneration: p.generation,
-		}
-
-		// If this component requires a cluster, connect to one
+		// Connect to cluster if a component requires it.
+		packageGeneration := 1
 		if component.RequiresCluster() {
```

Comment:
Took me some time to realize but we a have a fun little chicken and egg problem here. As we use an init package to setup a k3s cluster in some cases we cant just assume that the cluster we need to connect to exists if one of the components requires cluster access.

Instead we have to execute all of the component actions until we get to a point where the cluster is up and running and we can connect to it.